### PR TITLE
Login dropdown

### PR DESCRIPTION
**Pull request by jaypal412**
Ticket:https://www.pivotaltracker.com/story/show/158024365

Ticket: https://www.pivotaltracker.com/story/show/158024172

## Why?

**Feature** : Implement login drop down .
**Feature**: When there is no product or any UI list. Footer comes at the bottom of the screen. 

**Bug**: Dropdown is not showing correct in device view
**Bug**: Footer is coming top when there is no content in the body

## This change addresses the need by:

Made changes using the media query and the CSS required for the login dropdown

To achieve this I have made changes in the all body CSS and CSS of a footer.

[delivers #158024172]

[delivers #158024365]

### CODE REVIEW

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    </ul>
+  </div>
+  <div class="p-0 no-gutters" class="sub-sub-catgory" [@subCatgory]="stateName1">
+    <ul *ngIf="subChild">
+        <li (click)="childBack()">Back</li>
+      <li *ngFor="let taxon of subChild.taxons; let i= index">{{taxon.name}}</li>
+    </ul>
+  </div>
+</div> -->
+
```
Comment (gopalshimpi) [inline src/app/home/filter-mobile-menu/filter-mobile-menu.component.html]: @jaypal412 why this code is commented?

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+<input name="options" [(ngModel)]="options" type="radio" [value]="false" [checked]="!options"/> No
+<h2 *ngIf="options">Supply</h2>
+<h2 *ngIf="!options">Demand</h2>
+
+
+<accordion [closeOthers]="true">
+  <accordion-group>
+    <div accordion-heading class="clearfix">
+      I can have markup!
```
Comment (gopalshimpi) [inline src/app/home/filter-mobile-menu/filter-mobile-menu.component.html]: Can you please remove these hard coded values ` I can have markup!` and many morre like this in this `<accordion>` tag.

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  showCategory(taxon) {   
+    this.menuTaxons = taxon.taxons;   
+    console.log(taxon);
+    console.log(this.menuTaxons)
+  }
+  get stateName() {
+    return this.showParrent ? 'show' : 'hide'
+  }
+  
+  get stateName1() {
```
Comment (gopalshimpi) [inline src/app/home/filter-mobile-menu/filter-mobile-menu.component.ts]: you can change this to proper variable name. `stateName1`

File: src/app/home/home.component.ts [hunk 4]
```diff
@@ -43,9 +51,31 @@ export class HomeComponent implements OnInit {
         this.products = data
         if (this.products.count) { this.isProducts = true } else { this.isProducts = false }
       })
+      this.store.select(getTaxonomies)
+      .subscribe(data => {
+        this.fillterList = data; 
+        console.log(this.fillterList);
```
Comment (gopalshimpi) [inline src/app/home/home.component.ts]: Remove this console log.

File: src/app/home/sidebar/categories/categories.component.html [hunk 5]
```diff
@@ -24,7 +24,7 @@ <h4 *ngIf="categoryLevel.length > 0"> Catgeories</h4>
     <li *ngFor="let levels of categoryLevel; let index =i" (click)="emitSelection()">
       <a [routerLink]="['.']" class="list-group-item list-group-item-action" [queryParams]="{'q[name_cont]': levels.name, id: levels.id}">
        <span class="maninname">
-          {{levels.name}}
+          {{levels.c}}
```
Comment (gopalshimpi) [inline src/app/home/sidebar/categories/categories.component.html]: Why did you changed ? {{levels.name}} to {{level.c}}

File: src/app/layout/header/components/category-mobile-menu/category-mobile-menu.component.ts [hunk 6]
```diff
@@ -40,41 +40,41 @@ export class CategoryMobileMenuComponent implements OnInit {
...
+  showCategory(i) {
+    this.menuTaxons = this.taxonomies[0].root.taxons[i];
+    this.showParrent = !this.showParrent;
   }
   get stateName() {
     return this.showParrent ? 'show' : 'hide'
   }
-  
+
   get stateName1() {
```
Comment (gopalshimpi) [inline src/app/layout/header/components/category-mobile-menu/category-mobile-menu.component.ts]: Change variable names.

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.html (hunk 1 above)
Comment (jaypal412) [reply src/app/home/filter-mobile-menu/filter-mobile-menu.component.html]: Work in progress on that filter mobile menu

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.html (hunk 2 above)
Comment (jaypal412) [reply src/app/home/filter-mobile-menu/filter-mobile-menu.component.html]: Work in progress on that filter mobile menu

File: src/app/home/filter-mobile-menu/filter-mobile-menu.component.ts (hunk 3 above)
Comment (jaypal412) [reply src/app/home/filter-mobile-menu/filter-mobile-menu.component.ts]: Work in progress on that filter mobile menu

File: src/app/home/home.component.ts (hunk 4 above)
Comment (jaypal412) [reply src/app/home/home.component.ts]: Work in progress on that filter mobile menu

File: src/app/home/sidebar/categories/categories.component.html (hunk 5 above)
Comment (jaypal412) [reply src/app/home/sidebar/categories/categories.component.html]: This done by mistake. Change reverted.

**Pull request merged**